### DISCUSSION TITLE
New Status Map on d3.js

### DISCUSSION THREAD

**Discussion by white-line** [sni/thruk#1083] [Ideas]
https://nagioscore.demos.nagios.com/nagios/map.php?host=all as example ( login nagiosadmin, password nagiosadmin ). It is very comfortable. Especially balanced tree ( vertical ) mode, even more when 1000+ hosts are monitored.
In the meantime, we have to combine thruk and nagios 4 UI.
I would be grateful if someone implements such a map directly in thruk.

**Comment by sni** [comment]
move issue to disussions...

**Comment by white-line** [comment]
![nice](https://user-images.githubusercontent.com/13614853/112397467-33547480-8d13-11eb-8aca-da61a08e2219.png)
![very_nice](https://user-images.githubusercontent.com/13614853/112397478-37809200-8d13-11eb-91f7-f64881cfca9e.png)

**Comment by mintsoft** [comment]
Is the map in thruk a straight copy/port of the one from nagios3?

**Reply by white-line** [reply]
No, If you think about bottom screen. This map - is Nagios 4 new feature, from version 4.1.0 ( I am not sure ). If you want probe this map in your Thruk interface, you need install nagios 4 with UI. For example:
- Install Nagios 4 via "apt-get" utility in Debian ( 4.3.4 version in "Buster" repository )
- Install mk_livestatus module for Nagios 4 ( require compilation for Debian 10 buster ( i not found ready-made .deb package with livestatus ))
- Configure new backend in Thruk for new Nagios 4.
- Add or export hosts to Nagios 4.
- Check access for Nagios 4 UI. Maybe need edit Nagios 4 cgi.cfg.

For actions "Locate host on statusmap", add action-menu items in Thruk. For examlpe in thruk_local.conf:
```
<action_menu_items>
	hostmenu = {\
	   "icon": "/thruk/themes/{{theme}}/images/dropdown.png",\
	   "title": "Host Menu",\
	   "menu": [\
	       {\
		   "icon": "/thruk/themes/{{theme}}/images/status3.gif",\
		   "label": "Locate on Nagios Map",\
		   "action": "<your web server>/nagios4/map.php?host=$HOSTNAME$",\
		   "target": "main"\
	       },\
	   ]\
	}
</action_menu_items>

<action_menu_apply>
    hostmenu    = .*;$ 
</action_menu_apply>
```

New status map should open in new frame, or in new tab\window if `use_frames=0` in your thruk/cgi.conf.

**Comment by white-line** [comment]
I ported map from Nagios 4 to Thruk. See here [thruk_nagios4_features](https://github.com/white-line/thruk_nagios4_features)

**Reply by sistemmsn** [reply]
is development still alive?

**Reply by white-line** [reply]
> is development still alive?

I tested plugin in OMD 5.50 ( version of Thruk - 3.20.2 ) - it working fine. About development: me interesting "map" feature only - i realized it and my task done.

**Reply by sistemmsn** [reply]
Yes it is working thanks

**Reply by sstefanov** [reply]
@white-line, how you install it in OMD? Could you post some instructions, please?

**Reply by sistemmsn** [reply]
Just download the ZIP file. and add the code above to add to your menu.

**Reply by white-line** [reply]
> @white-line, how you install it in OMD? Could you post some instructions, please?

After you have created a new OMD instance:

```
sudo su -
# 
# go to "thruk/plugins-enabled" dir in your instance
cd /omd/sites/<your site>/etc/thruk/plugins-enabled
# where "your site" - OMD instance name
# 
# download zip file with plugin project
wget https://github.com/white-line/thruk_nagios4_features/archive/refs/heads/main.zip
# decompress file and move plugin dir
unzip main.zip
mv thruk_nagios4_features-main/nagios4_features/ ./
#
# delete unnecessary files
rm -rf thruk_nagios4_features-main
rm main.zip
# 
# restart yout OMD instance
# omd restart <your site>
```

Additional instruction about "action menu" and other configuration - see in README.md on [nagios4_features](https://github.com/white-line/thruk_nagios4_features)
